### PR TITLE
More unified search preperations

### PR DESCRIPTION
**Pull request by nickvergessen**
* Set the "searching from" url also on the search request, not only the provider gathering
* Don't show duplicate types in the search input: "Search apps, messages, messages, …"
* Fix double space in language
* Use the search result name for the header instead of the type name from the provider list (as it can have more context added)
* Don't lowercase types in the search input as it breaks some languages

PR is targetting #22111

### CODE REVIEW

File: core/src/views/UnifiedSearch.vue [hunk 1]
```diff
@@ -285,13 +287,15 @@ export default {
 			this.focusInput()
 			// Update types list in the background
 			this.types = await getTypes()
+			this.originalTypes = this.types
```
Comment (skjnldsv) [inline core/src/views/UnifiedSearch.vue:290]: Why is this needed?
I don't see why we have a duplicate updated list of providers?

Comment (nickvergessen) [reply core/src/views/UnifiedSearch.vue:290]: See below:

```js

				// Search providers can return a different name on the result,
				// so we overwrite it and use it for the header of the result list.
				// But when the search is reset, the original name is used again.
				this.types[key].name = request.data.name
```

Comment (skjnldsv) [reply core/src/views/UnifiedSearch.vue:290]: Why not use it while itterating from the results directly then ?

Comment (skjnldsv) [reply core/src/views/UnifiedSearch.vue:290]: https://github.com/nextcloud/server/blob/5a912c6bb2eadbdf44e18d89223230335b4889fc/core/src/views/UnifiedSearch.vue#L177-L178

Comment (skjnldsv) [reply core/src/views/UnifiedSearch.vue:290]: https://github.com/nextcloud/server/blob/5a912c6bb2eadbdf44e18d89223230335b4889fc/core/src/views/UnifiedSearch.vue#L362

Comment (skjnldsv) [reply core/src/views/UnifiedSearch.vue:290]: We could also just store the results like they're being sent to us by the api and adapt the other data ?

Comment (skjnldsv) [reply core/src/views/UnifiedSearch.vue:290]: :shrug:

Comment (nickvergessen) [reply core/src/views/UnifiedSearch.vue:290]: > Why not use it while itterating from the results directly then ?

Right, let me look into this.

Comment (nickvergessen) [reply core/src/views/UnifiedSearch.vue]: Done, please check if that is what you ment.

**Review by rullzer — APPROVED** [review]
sure
